xrmocap/ops/fourd_association/fourd_associator.py

```diff
@@ -0,0 +1,185 @@
+# yapf: disable
+import logging
+import numpy as np
+from typing import List, Tuple, Union
+import json
+import cv2
+from xrprimer.data_structure.camera import (
+    FisheyeCameraParameter, PinholeCameraParameter,
+)
+from xrprimer.utils.log_utils import get_logger
+
+from xrmocap.data_structure.keypoints import Keypoints
+
+from xrmocap.ops.triangulation.builder import (
+    BaseTriangulator, build_triangulator,
+)
+
+from xrmocap.ops.fourd_association.matching.builder import build_matching
+# yapf: enable
+
+
+class Camera():
+    def __init__(self,cam_param) -> None:
+        super().__init__()
+        self.originK = np.zeros((3,3), dtype=np.float)
+        self.cvR = np.zeros((3,3), dtype=np.float)
+        self.cvT = np.zeros(3, dtype=np.float)
+
+        self.originK = cam_param.intrinsic33()
+        self.cvT = np.array(cam_param.get_extrinsic_t())
+        self.cvR = np.array(cam_param.get_extrinsic_r())
+        self.imgSize = [cam_param.width, cam_param.height]
+        self.rectifyAlpha = 0
+        # self.cvK, _ = cv2.getOptimalNewCameraMatrix(self.originK,self.distCoeff, self.imgSize, self.rectifyAlpha)
+        self.cvK = self.originK
+
+        self.cvKi = np.linalg.inv(self.cvK)
+        self.cvRt = self.cvR.T
+        self.cvRtKi = np.matmul(self.cvRt, self.cvKi)
+        self.cvPos = -np.matmul(self.cvRt, self.cvT) 
+        
+        self.cvProj = np.zeros((3,4), dtype=np.float)
+        for i in range(3):
+            for j in range(4):
+                self.cvProj[i,j] = self.cvR[i,j] if j < 3 else self.cvT[i]
+        self.cvProj = np.matmul(self.cvK, self.cvProj)
+
+    def calcRay(self, uv):
+        ver  = -self.cvRtKi.dot(np.append(uv, 1).T)
+        return ver / np.linalg.norm(ver)
+        
+def convert_kps3d(multi_kps3d):
+    kps_arr = np.zeros((1,len(multi_kps3d),14,4))
+    for index, skel15 in enumerate(multi_kps3d):
```

Comment:
Rather than `skel15`, it is better to use some name like `openpose15_kps`, `coco15_kps` or `fourdag_kps`, and the convention should be found in our keypoints factory.